### ISSUE TITLE
Hi-BEHRT: Hierarchical Transformer-based model for accurate prediction of clinical events using multimodal longitudinal electronic health records

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#169]
## Abstract & Introduction
* processing long sequences에서 문제가 있음.
  * 전체 medical 기록을 capture하면 정확도가 높아질 것이라 생각하지만 수십년 간의 데이터를 집어넣으면 receptive field가 exceed됨. --> long-term dependencies를 잃게 됨.
  * gap을 다루기 위해 Hi-BEHRT를 제안함.
  * transformer의 receptive field of Transformers를 확장해서 더 긴 sequence로부터 association을 추출함.
* 기존 BEHRT의 성능을 뛰어 넘음.
* hierarchical Transformer에 대한 pre-training이 제대로 되어있지 않음.
* --> 본 논문에서 효과적인 end-to-end contrastive pre-training 전략을 제공함.
* adapting risk prediction model
* -->handle patients with thousands of records or even longer EHR sequence and avoid missing important historical information

### Data source and cohort selection
* Clinical Practice Research Datalink (CPRD) 사용.
* Hospital Episodes Statistics Admitted Patient Care data &death registration data from the Office for National Statistics
* split them into 60%, 10%, and 30% for training, tuning, and validation for the risk prediction task
* All the records before the baseline date were used as learning period to predict the 5-year risk of an event of interest after it
* ignored the events that occurred within 1-year after the baseline date
* randomly selected a baseline date within a 1-year to 5-year window before the incident date
* excluded patients who had less than 3 years of records -->  risk prediction task의 경우.

### Case identification & Data processing
* focused on 5-year risk estimation of the incidence of HF, diabetes, CKD, and stroke
* 환자 분석
  <image src="https://user-images.githubusercontent.com/67627471/134812150-a60d180f-d395-440a-97c9-2afaa1d15a2c.png" width="40%" height="40%" />  <image src="https://user-images.githubusercontent.com/67627471/134812160-0cc9cf1a-08bd-4c52-a6a0-363c79742843.png" width="40%" height="40%" />
* included records from diagnoses, medications, hospital procedures, GP tests, BP measurements (both systolic & diastolic pressure), drinking status, smoking status, BMI
* 진단에는 ICD-10을 standard format으로 이용.
* mapped all diagnostic records from the primary care
* additional feature of each record, age, was also calculated by the event date and the patient’s date of birth

### Pre-training
* 기존 BEHRT와 새로운 hierarchical BEHRT 구조 비교.
  <image src="https://user-images.githubusercontent.com/67627471/134811770-b3003ae6-c24d-4374-8646-7001bea81cd0.png" width="100%" height="100%" />

#### BYOL for hierarchical BEHRT
* applied BYOL for the self-supervised pretraining --> MLM과 같이 사용.
* hierarchical BEHRT에 적용된 BYOL 아이디어. 
  <image src="https://user-images.githubusercontent.com/67627471/134811853-ab1e3f39-9787-432a-980d-a872e40d7177.png" width="70%" height="70%" />
* 기존 BYOL에는 data augmentation, encoder, projector, predictor가 있음.
  <image src="https://user-images.githubusercontent.com/67627471/134811938-cd014bc3-c541-47bb-a84c-67684ac655a8.png" width="50%" height="50%" />
* encoder : Hi-BEHRT model
* projector & predictor : hidden layer multi-layer perceptron network
* data augmentation : 기존 구조에 two augmentation strategies 추가함 (아래)

#### EHR augmentation & Segment augmentation
* EHR augmentation : applied before sliding window
  * applied random crop for the EHR with a probability 𝑃_c
  * randomly masked each token with a probability 𝑃_m
* Segment augmentation : applied to segment representations before feeding them into feature aggregato
  * masked a segment representation by casting it to 0 or added a Gaussian noise to deprecate the latent representation

## Results
* Evaluation on risk prediction
  <image src="https://user-images.githubusercontent.com/67627471/134812264-562a441f-254b-49f0-9efd-e3e5191cd644.png" width="50%" height="50%" />
* Semi-supervised training for risk prediction
  <image src="https://user-images.githubusercontent.com/67627471/134812288-8f644b70-eca2-4cbf-8604-92ed2107fdf8.png" width="50%" height="50%" />

##
* transformer 논문을 꽤 읽었는데 라벨을 처음부터 만들어야 했나..
